sdk/communication/communication-administration/src/phoneNumber/lro/purchase/operation.ts

```diff
@@ -0,0 +1,99 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import { AbortSignalLike } from "@azure/abort-controller";
+import { PollOperationState, PollOperation } from "@azure/core-lro";
+import { PhoneNumberSearch } from "../../generated/src/models";
+import { BeginPurchaseSearchOptions, PhoneNumberPollerClient } from "../../models";
+import { isComplete } from "../util";
+
+export interface PurchaseSearchPollOperationState extends PollOperationState<PhoneNumberSearch> {
+  /**
+   * The id of the search returned by createSearch.
+   */
+  searchId: string;
+
+  /**
+   * Represents the poller client used internally.
+   */
+  client: PhoneNumberPollerClient;
+  /**
+   * Options for refreshing a search.
+   */
+  options?: BeginPurchaseSearchOptions;
+}
+
+/**
+ * Represents the refresh poll operation.
+ */
+export interface PurchaseSearchPollOperation
+  extends PollOperation<PurchaseSearchPollOperationState, PhoneNumberSearch> {}
+
+async function update(
+  this: PurchaseSearchPollOperation,
+  options: {
+    abortSignal?: AbortSignalLike;
+    fireProgress?: (state: PurchaseSearchPollOperationState) => void;
+  } = {}
+): Promise<PurchaseSearchPollOperation> {
+  const state = this.state;
+  const { searchId, client } = state;
+  const requestOptions = state.options || {};
+
+  if (options.abortSignal) {
+    requestOptions.abortSignal = options.abortSignal;
+  }
+
+  try {
+    if (!state.isStarted) {
+      await client.purchaseSearch(searchId, requestOptions);
+      state.isStarted = true;
+      state.result = await client.getSearch(searchId, requestOptions);
```

Comment:
Could you remove these two lines? And let the if (!state.isCompleted) block take care of it?

Same comment applies to the other operations.

Reply:
I'm not quite following, why can we remove these lines?

Reply:
I guess there is an implicit assumption. If !state.isStart is true, then it is implied that !state.isCompleted is also true.

Hence, lines 56,57, which is identical to lines 51,52, will get executed.